examples/adiabatic/linear.py

```diff
@@ -38,13 +39,14 @@ def main(nqubits, hfield, T, dt, solver, save):
     """
     h0 = hamiltonians.X(nqubits)
     h1 = hamiltonians.TFIM(nqubits, h=hfield)
+    bac = h1.backend
 
     # Calculate target values (H1 ground state)
     target_state = h1.ground_state()
-    target_energy = h1.eigenvalues()[0].numpy().real
+    target_energy = bac.to_numpy(h1.eigenvalues()[0]).real
 
     # Check ground state
-    state_energy = callbacks.Energy(h1)(target_state).numpy()
+    state_energy = bac.to_numpy(callbacks.Energy(h1).apply(bac, target_state)).real
     np.testing.assert_allclose(state_energy.real, target_energy)
```

Comment:
Also, if you change this to
```suggestion
    h1.backend.assert_allclose(state_energy, target_energy)
```
I think you can avoid the `bac.to_numpy` calls.

Reply:
While for the `state_energy` I can do without, the `target_energy` will need the `to_numpy`/`.real` anyway due to it being used later on by matplotlib so that it works for any backend.

Any suggestions? (I can do `state_energy` without `to_numpy` and `target_energy` with for instance).

Reply:
It should be fine to leave `to_numpy` then. Indeed the plotting might fail for the cupy backend if we don't convert to numpy first, so it should be safe to use `to_numpy`.